test/unit/spec/PrimaryAuth_spec.js

```diff
@@ -1996,56 +2000,119 @@ function (Q, OktaAuth, LoginUtil, Okta, Util, AuthContainer, PrimaryAuthForm, Be
-          expect(test.form.facebookButton().length).toBe(1);
-          expect(test.form.googleButton().length).toBe(1);
-          expect(test.form.microsoftButton().length).toBe(1);
-          expect(test.form.linkedInButton().length).toBe(1);
+          var buttons = test.form.socialAuthButtons();
+          expect(buttons.eq(0).hasClass('social-auth-linkedin-button')).toBe(true);
+          expect(buttons.eq(1).hasClass('social-auth-facebook-button')).toBe(true);
+          expect(buttons.eq(2).hasClass('social-auth-google-button')).toBe(true);
+          expect(buttons.eq(3).hasClass('social-auth-microsoft-button')).toBe(true);
         });
       });
-      itp('shows idps in the order specified', function () {
+      itp('optionally adds a class for idp buttons', function () {
         var settings = {
           idps: [
             {
-              type: 'LINKEDIN',
-              id: '0oaidiw9udOSceD1111'
+              type: 'GOOGLE',
+              id: '0oaidiw9udOSceD5678',
+              className: 'example-class'
             },
             {
               type: 'FACEBOOK',
-              id: '0oaidiw9udOSceD1234'
+              id: '0oaidiw9udOSceD1234',
             },
             {
-              type: 'GOOGLE',
-              id: '0oaidiw9udOSceD5678'
-            },
-            {
-              type: 'MICROSOFT',
-              id: '0oaidiw9udOSceD3333'
+              id: '0oaidiw9udOSceD1111',
+              text: 'Not default text',
+              className: 'other-class'
             }
           ]
         };
-        return setup(settings).then(function (test) {
+        return setup(settings).then(function (test){
           var buttons = test.form.socialAuthButtons();
-          expect(buttons.eq(0).hasClass('social-auth-linkedin-button')).toBe(true);
+          expect(buttons.eq(0).hasClass('social-auth-google-button')).toBe(true);
+          expect(buttons.eq(0).hasClass('social-auth-facebook-button')).toBe(false);
+          expect(buttons.eq(0).hasClass('example-class')).toBe(true);
+          expect(buttons.eq(0).hasClass('other-class')).toBe(false);
+
+          expect(buttons.eq(1).hasClass('social-auth-google-button')).toBe(false);
           expect(buttons.eq(1).hasClass('social-auth-facebook-button')).toBe(true);
-          expect(buttons.eq(2).hasClass('social-auth-google-button')).toBe(true);
-          expect(buttons.eq(3).hasClass('social-auth-microsoft-button')).toBe(true);
+          expect(buttons.eq(1).hasClass('example-class')).toBe(false);
+          expect(buttons.eq(1).hasClass('other-class')).toBe(false);
+
+          expect(buttons.eq(2).hasClass('social-auth-google-button')).toBe(false);
+          expect(buttons.eq(2).hasClass('social-auth-facebook-button')).toBe(false);
+          expect(buttons.eq(2).hasClass('example-class')).toBe(false);
+          expect(buttons.eq(2).hasClass('other-class')).toBe(true);
+        });
+      });
+      itp('displays generic idp buttons', function () {
```

Comment:
nit:
more explicitly like `displays generic idp buttons for not default supported type`?